seismic/stream_processing.py

```diff
@@ -52,3 +54,72 @@ def sinc_resampling(t, y, t_new):
     y_new = np.matmul(np.sinc((Ts.T - T.T) / dt), y)
     return y_new
 # end func
+
+
+def back_azimuth_filter(baz, baz_range):
+    """Check if back azimuth `baz` is within range. Inputs must be in the range [0, 360] degrees.
+
+    :param baz_range: Pair of angles in degrees.
+    :type baz_range: List or array of 2 floats, min and max back azimuth
+    :return: True if baz is within baz_range, False otherwise.
+    """
+    assert not np.any(np.isinf(np.array(baz_range)))
+    assert not np.any(np.isnan(np.array(baz_range)))
+    assert 0 <= baz <= 360
+    assert 0 <= baz_range[0] <= 360
+    assert 0 <= baz_range[1] <= 360
+    baz_range = list(copy.copy(baz_range))
+    while baz_range[0] > baz_range[1]:
+        baz_range[0] -= 360
+    return ((baz_range[0] <= baz <= baz_range[1]) or
+            (baz_range[0] <= baz - 360 <= baz_range[1]) or
+            (baz_range[0] <= baz + 360 <= baz_range[1]))
+# end func
+
+
```

Comment:
Below here, helper function to test effects of perturbing channel data. This will be used in follow-up tickets, and can be used via the "custom_preproc" fields in the json configuration file.